# **Exploring the Indian Fitness Tracker Market: Demand, Trends, and Price Correlations**


In [17]:
import pandas as pd

In [18]:
data_path = './data/smartwatches.csv'

In [19]:
df = pd.read_csv(data_path)

In [20]:
df.drop_duplicates(inplace=True)

to read the data

df = data frame

In [21]:
df.head()

Unnamed: 0       Brand  Current Price  Original Price  Discount Percentage  \
0           0       noise        82990.0         89900.0             7.686318   
1           1  fire-boltt         3799.0         16999.0            77.651627   
2           2        boat         1999.0          7990.0            74.981227   
3           3  fire-boltt         1799.0         19999.0            91.004550   
4           4       noise         1599.0          4999.0            68.013603   

   Rating  Number OF Ratings                              Model Name  \
0     4.0               65.0   Wrb-sw-colorfitpro4alpha-std-rgld_pnk   
1     4.3            20788.0                                  BSW046   
2     3.8            21724.0                               Wave Call   
3     4.3            13244.0                                  BSW053   
4     4.1            13901.0  Wrb-sw-colorfitpulsegobuzz-std-blk_blk   

  Dial Shape Strap Color Strap Material Touchscreen  Battery Life (Days)  \
0        NaN         NaN            NaN         NaN                  8.0   
1        NaN         NaN        Silicon         Yes                  3.5   
2        NaN         NaN        Silicon         Yes                  8.0   
3        NaN         NaN        Silicon         Yes                  3.5   
4        NaN         NaN          Other         Yes                  8.0   

  Bluetooth Display Size     Weight  
0       Yes          NaN  35 - 50 g  
1       Yes   1.8 inches  50 - 75 g  
2       Yes   1.7 inches  35 - 50 g  
3       Yes   1.8 inches      75g +  
4       Yes   1.7 inches  35 - 50 g

give a header of the data. first 5 values.

1. Unnamed: 0 (remove)

In [22]:
df.shape

(450, 16)

there are 450 rows and 16 columns in the dataset

In [23]:
train_df = df.sample(frac=0.8, random_state=42)

In [24]:
train_df = train_df.reset_index()

In [25]:
test_df = test_df.reset_index()

In [26]:
test_df = df.drop(train_df.index)

In [27]:
print(train_df.shape, test_df.shape)

(360, 17) (90, 16)


In [28]:
df.tail()

Unnamed: 0       Brand  Current Price  Original Price  \
445         445  fire-boltt         5999.0          8999.0   
446         446  fire-boltt         3700.0          9999.0   
447         447  fire-boltt         5999.0          8999.0   
448         448  fire-boltt         3200.0          7999.0   
449         449  fire-boltt         3000.0          7999.0   

     Discount Percentage  Rating  Number OF Ratings Model Name Dial Shape  \
445            33.337037     2.5                NaN     bsw003     Circle   
446            62.996300     NaN                NaN  beast pro     Square   
447            33.337037     2.5                NaN     bsw003     Circle   
448            59.994999     NaN                NaN     call 2     Square   
449            62.495312     NaN                NaN     call 2     Square   

    Strap Color Strap Material Touchscreen  Battery Life (Days) Bluetooth  \
445        Grey        Silicon         Yes                 22.0       Yes   
446       Black        Silicon         Yes                  8.0       Yes   
447        Gold        Silicon         Yes                 22.0       Yes   
448        Blue        Silicon         Yes                 22.0       Yes   
449       Black        Silicon         Yes                 22.0       Yes   

    Display Size Weight  
445   1.3 inches    NaN  
446   0.1 inches    NaN  
447   1.3 inches    NaN  
448   0.1 inches    NaN  
449   0.1 inches    NaN

tail (last 5 values of the data)

In [29]:
train_df.dtypes

index                    int64
Unnamed: 0               int64
Brand                   object
Current Price          float64
Original Price         float64
Discount Percentage    float64
Rating                 float64
Number OF Ratings      float64
Model Name              object
Dial Shape              object
Strap Color             object
Strap Material          object
Touchscreen             object
Battery Life (Days)    float64
Bluetooth               object
Display Size            object
Weight                  object
dtype: object

for knowing the datatypes of various values present in the data

In [30]:
train_df.isna().sum()

index                    0
Unnamed: 0               0
Brand                    0
Current Price            6
Original Price          56
Discount Percentage     56
Rating                   4
Number OF Ratings       45
Model Name              30
Dial Shape             100
Strap Color            100
Strap Material          56
Touchscreen             31
Battery Life (Days)     30
Bluetooth                5
Display Size            27
Weight                 149
dtype: int64

In [31]:
train_df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 360 entries, 0 to 359
Data columns (total 17 columns):
 #   Column               Non-Null Count  Dtype  
---  ------               --------------  -----  
 0   index                360 non-null    int64  
 1   Unnamed: 0           360 non-null    int64  
 2   Brand                360 non-null    object 
 3   Current Price        354 non-null    float64
 4   Original Price       304 non-null    float64
 5   Discount Percentage  304 non-null    float64
 6   Rating               356 non-null    float64
 7   Number OF Ratings    315 non-null    float64
 8   Model Name           330 non-null    object 
 9   Dial Shape           260 non-null    object 
 10  Strap Color          260 non-null    object 
 11  Strap Material       304 non-null    object 
 12  Touchscreen          329 non-null    object 
 13  Battery Life (Days)  330 non-null    float64
 14  Bluetooth            355 non-null    object 
 15  Display Size         333 non-null    obj

In [32]:
numerical_data = [feature for feature in train_df.columns if train_df[feature].dtype != 'object']

In [33]:
continuous_data = [feature for feature in train_df.columns if train_df[feature].dtype == 'object']

In [34]:
numerical_data

['index',
 'Unnamed: 0',
 'Current Price',
 'Original Price',
 'Discount Percentage',
 'Rating',
 'Number OF Ratings',
 'Battery Life (Days)']

In [35]:
continuous_data

['Brand',
 'Model Name',
 'Dial Shape',
 'Strap Color',
 'Strap Material',
 'Touchscreen',
 'Bluetooth',
 'Display Size',
 'Weight']